src/TsBodyPrinter.ts

```diff
@@ -99,6 +105,9 @@ export default class TsBodyPrinter {
     this.printConstructor();
     this.printProperties();
     this.printMethods();
+    if (this.buildType === BuildType.base) {
+      this.printPromiseLike();
+    }
```

Comment:
This seems a bit broad. For example, what happens when DOMParser (https://developer.mozilla.org/en-US/docs/Web/API/DOMParser) is awaited?

Should we limit this to only Node and it's descendants?

Reply:
That's how I did it originally. After our conversation when you said "everything" should be await-able, I took that literally... Node and it's descendants had a lot that didn't get covered, so I would just need help going through all the things that don't extend Node that still need awaited.